sdk/src/main/scala/com/horizen/account/state/CertificateKeyRotationMsgProcessor.scala

```diff
@@ -122,7 +123,13 @@ case class CertificateKeyRotationMsgProcessor(params: NetworkParams) extends Nat
 
     val signingKeyFromConfig = params.signersPublicKeys(index)
     val masterKeyFromConfig = params.mastersPublicKeys(index)
-    val newKeyAsMessage: Array[Byte] = keyRotationProof.newKey.getHash
+
+    val newKeyAsMessage = keyRotationProof.keyType match {
```

Comment:
Will put this comment here:
1) Let's initialize this MsgProccessor only in case of new cert circuit. No sense to keep it for the configuration with the old cert circuit. All the info is present in `params.circuitType`
2) For this we need to modify `MessageProcessorUtil.getMessageProcessorSeq` method only.
3) then check the tests and check sc_evm_settings.conf. Maybe some regression data will be outdated.